core/src/main/java/bio/terra/pearl/core/service/kit/KitRequestService.java

```diff
@@ -57,43 +60,40 @@ public KitRequestService(KitRequestDao dao,
         this.studyEnvironmentService = studyEnvironmentService;
         this.studyService = studyService;
         this.objectMapper = objectMapper;
+        this.daoUtils = daoUtils;
     }
 
     /**
      * Send a request for a sample kit to Pepper.
-     *
-     * 1. Create and save the sample kit request in the Juniper database
-     * 2. Send the sample kit request to Pepper
-     * 3. Record the status response from Pepper in the Juniper database
-     *
-     * These steps are all performed in a single transaction. However, each could fail independently. If there's an
-     * error from Pepper, then the sample kit request will not be saved in Juniper, and it's safe to assume that there
-     * is no new request in Pepper. If there's an error saving the status response from Pepper, then it's possible for
-     * there to be a request in Pepper that does not exist in Juniper. We could potentially split that step into its own
-     * transaction to help avoid that potential inconsistency.
+     * Throws PepperApiException if the Pepper API request failed
      */
-    @Transactional
-    public KitRequest requestKit(AdminUser adminUser, String studyShortcode, Enrollee enrollee, String kitTypeName) {
+    public KitRequest requestKit(AdminUser adminUser, String studyShortcode, Enrollee enrollee, String kitTypeName)
+    throws PepperApiException {
         // create and save kit request
-        Profile profile = profileService.loadWithMailingAddress(enrollee.getProfileId())
-                .orElseThrow(() -> new RuntimeException("Missing profile for enrollee: " + enrollee.getShortcode()));
+        if (enrollee.getProfileId() == null) {
+            throw new IllegalArgumentException("No profile for enrollee: " + enrollee.getShortcode());
```

Comment:
Agree -- also because this check likely will get finer-scale -- sure they have a profile, but what if they have a profile with no address?  or an address with no street?

Reply:
It's an IllegalArgumentException, which AbstractGlobalExceptionHandler will turn into a BAD_REQUEST response.  You're right that we probably want to catch this in the Ext service for multi-requests.  Will add.